### ISSUE TITLE
Issue with icon shapes and apps that can change their icons

### ISSUE THREAD

**Issue by Younes-L** [Deletescape-Media/Lawnchair#661]
Since I updated to 1.1.0.1233
- Combine Pixel style preference to icon shape on API24+

G Calendar switched to a round icon while all the other icons stayed the same.

I tried messing with the shape preferences, but "Do not change" doesn't seem to work properly anymore. It just goes to random circles here and there or retains the previous shape.

"None" breaks app that can change their icon. Basically when I try to edit and icon on the homescreen or drawer, it will be shown correctly but the one that ends up being displayed isn't the correct one (apps : flamingo, snapcamera HDR)

Right now I messed alot with shapes, rebuilt database multiple times, thankfully I was able to get rid of the circles except for Flamingo which default to white circle background, and snap (red gallery instead of blue camera, for some reason)

Device is Bacon/A0001 , 7.1.2 AOSP+CAF

Lawnchair build 1238

**Comment by Younes-L** [comment]
I think I have a less confusing description of the issue(s) now :

Setting icon shape to "none" will prevent apps that changes their icons to be displayed with the correct icon.

Setting icon shape to "do not change" does the same thing as "circle" , for some reason. I am not a Pixel user and I've been using all sort of launchers, including lawnchair, none of them enabled round icons by default. It seems to me that having round icons with icon shape set to "do not change" is a bug.

While icon shape is set to anything that isn't "none" :

  • apps that support adaptative will have the correct shape, 
  • apps that have a round icon will turn round, 
  • apps that only have 1 icon will keep displaying same icon
  • apps that can change their icon will finally display the correct icon, too bad it doesn't have to be round and all other apps will either be round or some other shape.


In conclusion I'm seeing 2 bugs here, "do not change" shouldn't be circle, and "none" breaks apps with multiple icons.

Screenshot(s) below :
____________________________________

This is do not change, displayed as circles, apps that can change their icons will display the correct icon (ICS camera for Snap and hexagon for Flamingo)
![](https://i.imgur.com/DE5lFIK.png)

____________________________________

This is none, apps have their default uniquely shaped icons that we all love, except for the apps that have multiple icons, which are displayed with the icons their appear with when seen under system settings or app info (flamingo and snap)
![](https://i.imgur.com/xlnSn8l.png)
   

Device is Oneplus A0001/Bacon, 7.1.2 AOSP+CAF, lawnchair build 1238
